src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy

```diff
@@ -67,9 +65,29 @@ class DockerJavaApplication {
      * The tag used for the Docker image.
      * <p>
      * Defaults to {@code <project.group>/<applicationName>:<project.version>}.
+     * @deprecated use {@link #tags} - will be removed in 6.0.0
      */
+    @Deprecated
     final Property<String> tag
 
+    /**
+     * The tag used for the Docker image.
+     * <p>
+     * Defaults to {@code [<project.group>/<applicationName>:<project.version>]}.
+     */
+    final ListProperty<String> tags
+
+    /**
+     * Tags that will be used for building the Docker image, but will not be pushed to the repo.
+     * In order for these tags to be built, they should be specified in the {@link #tags} property.
+     * e.g. for rapid local development, "busybox:latest" would be useful, but becomes confusing when
+     * pushed to a remote repo
+     * <p>
+     * Defaults to {@code []}
+     */
+    final ListProperty<String> localOnlyTags
```

Comment:
I don't think we should introduce this functionality. It makes understanding the API harder and users will likely get confused as they don't look the API. I can see your use case but why not go with a concrete tag (e.g. semantic versioning) and use that for local testing? I am just not a big fan of `latest` in Docker as it leads people to believe that it actually is the latest in terms of a SNAPSHOT.

Reply:
@bmuschko checkout the convo we had elsewhere. It wouldn't necessarily be for latest (once the name is changed) but for any tag we don't want pushed remotely. For example I create 10 tags but only want to push 5 of them for any number of reasons.

Reply:
@cdancy Really trying to understand the "reasons" and identify if there are other ways to make this happen without having to change the code as proposed. Can someone list the concrete use cases?

Reply:
May be naive at this point but I took this along the same lines as the change for `DockerSpringBootApplication` which is discussed further above.

Reply:
I guess you are referring to this?

> So I was thinking of my team's specific usecase, where we build ":latest" and reference it in all of our kubernetes manifests so that its fairly easy to do local development.

It doesn't really describe the use case. It only describes the "what" and not the "why". It will need more context for me to understand it. I am strongly voting for making the change for allowing building/pushing multiple tags first. After that we can have another discussion about which of those tags should be pushed and why. The changes in this PR try to implement multiple use cases and gets far too confusing to discuss.

Here's what I am trying to really figure out:

- What is the actual use case for not pushing all tags that were declared?
- Can this be solved in a different way without making these changes?
- Is this a common use case that would justify making the code more complex and harder to understand for other users? Maybe this use case is very specific to this one team and it might be better to let them build their own plugin.

Reply:
> What is the actual use case for not pushing all tags that were declared?

Balancing strict governance policies for Docker repos with developer productivity. The `:latest` tag is useful for doing local development with Helm and Kubernetes, so that we can test our manifests locally without having to edit the version every time. However, having a `:latest` tag is not allowed in our repo (everything must be versioned), and if we tried to push all of the tags, that one would be rejected and the whole task would fail.